### PR TITLE
Merge branch 'main' into dev-preview for version 1.27.1

### PR DESCRIPTION
**Pull request by royjit**
*Issue #, if available:*

*Description of changes:*

*Check points: (check or cross out if not relevant)*

- [ ] Added new tests to cover change, if needed
- [ ] Build succeeds with all target using Swift Package Manager
- [ ] All unit tests pass
- [ ] All integration tests pass
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [ ] PR title conforms to conventional commit style
- [ ] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/API/Sources/AWSAPIPlugin/Operation/AWSGraphQLSubscriptionOperation.swift [hunk 1]
```diff
@@ -173,18 +173,36 @@ final public class AWSGraphQLSubscriptionOperation<R: Decodable>: GraphQLSubscri
...
+
+            if graphQLErrors.hasUnauthorizedError() {
+                errorDescription += ": \(APIError.UnauthorizedMessageString)"
+            }
+
             let graphQLResponseError = GraphQLResponseError<R>.error(graphQLErrors)
             dispatch(result: .failure(APIError.operationError(errorDescription, "", graphQLResponseError)))
             finish()
             return
+        } else if case ConnectionProviderError.unauthorized = error {
```
Comment (lawmicha) [inline AmplifyPlugins/API/Sources/AWSAPIPlugin/Operation/AWSGraphQLSubscriptionOperation.swift:189]: this uses AppSyncRTC 2.0.0, need to verify the merge also upgrades AppSyncRTC

File: Package.resolved [hunk 2]
```diff
@@ -6,8 +6,8 @@
         "repositoryURL": "https://github.com/aws-amplify/aws-appsync-realtime-client-ios.git",
         "state": {
           "branch": null,
-          "revision": "28f61d7ca362699941cc8844d34d082f1eb49602",
-          "version": "1.10.0"
+          "revision": "36ed209d55aacc127356a387b51cef5f94ec9ac7",
+          "version": "2.0.0"
```
Comment (lawmicha) [inline Package.resolved:10]: hmm i don't see the corresponding Package.swift changes

Comment (lawmicha) [reply Package.resolved:10]: i think we need to add another commit to bump to 2.0.0 here https://github.com/aws-amplify/amplify-ios/blob/dev-preview/Package.swift#L11-L15

File: AmplifyPlugins/API/Sources/AWSAPIPlugin/Operation/AWSGraphQLSubscriptionOperation.swift (hunk 1 above)
Comment (royjit) [reply AmplifyPlugins/API/Sources/AWSAPIPlugin/Operation/AWSGraphQLSubscriptionOperation.swift:189]: I got a build failure, will update

File: Package.resolved (hunk 2 above)
Comment (royjit) [reply Package.resolved:10]: Updated the PR. fixes ARTC to latest version and build failures

**Review by lawmicha — DISMISSED** [review]

**Review by ruisebas — APPROVED** [review]

**Review by lawmicha — APPROVED** [review]

**Review by ameter — APPROVED** [review]

**Pull request merged**